lib/browser/api/extensions.js

```diff
@@ -79,8 +78,10 @@ var sendToBackgroundPages = function (extensionId, session, event) {
   var args = [].slice.call(arguments, 2)
   pages.forEach(function (backgroundPage) {
     try {
-      // only send to background pages in the same browser context
-      if (backgroundPage.session.equal(session)) {
+      var sendToAll = (typeof (session) === 'string' && session === 'all')
```

Comment:
this change makes sense from a code perspective, but sending to `all` is a very special case and I'm a little concerned that this will get used where it shouldn't be. Maybe we should at least add a comment here explaining?